### PR TITLE
fragment size and message size can be customized now.

### PR DESCRIPTION
**Pull request by jingtaozf**

### CODE REVIEW

File: hunchensocket.lisp [hunk 1]
```diff
@@ -1,5 +1,8 @@
 (in-package :hunchensocket)
 
+(defvar websocket-maximum-fragment-size #xffff) ;64KiB
```
Comment (joaotavora) [inline hunchensocket.lisp]: You should rename this `*websocket-maximum-fragment-size*` and use `defparameter`.

File: hunchensocket.lisp [hunk 2]
```diff
@@ -55,7 +58,7 @@
 (defclass websocket-resource ()
   ((clients :initform nil :reader clients)
    (client-class :initarg :client-class :initform 'websocket-client)
-   (lock :initform (make-lock))))
+   (lock :initform (make-lock "websocket"))))
```
Comment (joaotavora) [inline hunchensocket.lisp]: A better name for this lock (and indeed it is a good idea to name it) is "websocket-resource"

File: hunchensocket.lisp [hunk 3]
```diff
@@ -43,7 +46,7 @@
    (request    :initarg request
                :reader client-request
                :initform (error "Must make clients with requests"))
-   (write-lock :initform (make-lock))
+   (write-lock :initform (make-lock "websocket client"))
```
Comment (joaotavora) [inline hunchensocket.lisp]: "websocket client" -> "websocket-client write lock" please

File: hunchensocket.lisp [hunk 4]
```diff
@@ -167,6 +170,8 @@ format control and arguments."
   (with-slots (clients lock) resource
     (unwind-protect
          (progn
+           #+sbcl(sb-ext:atomic-push client clients)
```
Comment (joaotavora) [inline hunchensocket.lisp]: nice!

File: hunchensocket.lisp [hunk 5]
```diff
@@ -439,11 +444,11 @@ payloads."
     (:rfc-6455
      (handler-bind ((websocket-error
                       #'(lambda (error)
-                          (with-slots (status format-control format-arguments)
+                          (with-slots (error-status format-control format-arguments)
```
Comment (joaotavora) [inline hunchensocket.lisp]: Can you explain this part?  Does `websocket-error` have both slots `status` *and* `error-status`?  I don't remember anymore to be honest.

File: package.lisp [hunk 6]
```diff
@@ -9,6 +9,8 @@
    ;; acceptor classes
    #:websocket-acceptor
    #:websocket-ssl-acceptor
+   #:websocket-maximum-fragment-size
```
Comment (joaotavora) [inline package.lisp]: You'll need to update this to the new `*websocket...*` names.

File: hunchensocket.lisp [hunk 7]
```diff
@@ -82,9 +85,9 @@
   (:method ((resource websocket-resource)
             (client websocket-client) opcode length total)
     (declare (ignore resource client))
-    (cond ((> length #xffff) ; 65KiB
+    (cond ((> length websocket-maximum-fragment-size)
```
Comment (joaotavora) [inline hunchensocket.lisp]: You'll need to update this to the new `*websocket...*` names.

**Review by joaotavora — DISMISSED** [review]
Thank you very much.  But please take a look at the changes I requested.

File: hunchensocket.lisp [hunk 8]
```diff
@@ -170,6 +170,9 @@ format control and arguments."
   (with-slots (clients lock) resource
     (unwind-protect
          (progn
+           (format t "resource is ~s, lock is ~s~%" lock resource)
```
Comment (joaotavora) [inline hunchensocket.lisp]: Unsolicited stdout print?

Comment (joaotavora) [reply hunchensocket.lisp]: Never mind, I see you deleted it in another commit.

File: hunchensocket.lisp (hunk 5 above)
Comment (jingtaozf) [reply hunchensocket.lisp]: Yes, it has both slots but maybe I should not change it here.

Comment (joaotavora) [reply hunchensocket.lisp]: > Yes, it has both slots but maybe I should not change it here.

Yes, perhaps we can leave this for another pull request.  But can you remind me what the purpose of each one of those slots is?

File: hunchensocket.lisp (hunk 7 above)
Comment (jingtaozf) [reply hunchensocket.lisp]: fixed in new commit.

**Review by hanshuebner — APPROVED** [review]

**Pull request merged**